Code/Tools/SceneAPI/SceneCore/Utilities/SceneGraphSelector.cpp

```diff
@@ -40,18 +43,67 @@ namespace AZ
                 return object && object->RTTI_IsTypeOf(DataTypes::IMeshData::TYPEINFO_Uuid());
             }
 
-            Containers::SceneGraph::NodeIndex SceneGraphSelector::RemapToOptimizedMesh(const Containers::SceneGraph& graph, const Containers::SceneGraph::NodeIndex& index)
+            Containers::SceneGraph::NodeIndex SceneGraphSelector::RemapToOptimizedMesh(
+                const Containers::SceneGraph& graph, const Containers::SceneGraph::NodeIndex& index)
             {
-                const auto& nodeName = graph.GetNodeName(index);
-                const AZStd::string optimizedName = AZStd::string(nodeName.GetPath(), nodeName.GetPathLength()).append(OptimizedMeshSuffix);
-                if (auto optimizedIndex = graph.Find(optimizedName); optimizedIndex.IsValid())
+                return RemapNodeIndex(graph, index, SceneAPI::Utilities::OptimizedMeshPropertyMapKey);
+            }
+
+            Containers::SceneGraph::NodeIndex SceneGraphSelector::RemapToOriginalUnoptimizedMesh(
+                const Containers::SceneGraph& graph, const Containers::SceneGraph::NodeIndex& index)
+            {
+                return RemapNodeIndex(graph, index, SceneAPI::Utilities::OriginalUnoptimizedMeshPropertyMapKey);
+            }
+
+            Containers::SceneGraph::NodeIndex SceneGraphSelector::RemapNodeIndex(
+                const Containers::SceneGraph& graph,
+                const Containers::SceneGraph::NodeIndex& index,
+                const AZStd::string_view customPropertyKey)
+            {
+                // Search the immediate children for an ICustomPropertyData node to lookup the optimized mesh index
+                const Containers::SceneGraph::NodeIndex customPropertyIndex =
+                    GetImmediateChildOfType(graph, index, azrtti_typeid<AZ::SceneAPI::DataTypes::ICustomPropertyData>());
+
+                if (customPropertyIndex.IsValid())
                 {
-                    return optimizedIndex;
+                    const DataTypes::ICustomPropertyData* customPropertyDataNode =
+                        azrtti_cast<const DataTypes::ICustomPropertyData*>(graph.GetNodeContent(customPropertyIndex).get());
+
+                    // Now look up the optimized index
+                    const auto& propertyMap = customPropertyDataNode->GetPropertyMap();
+                    const auto iter = propertyMap.find(customPropertyKey);
+
+                    if (iter != propertyMap.end())
+                    {
+                        const AZStd::any& optimizedNodeIndex = iter->second;
```

Comment:
It seems the node index isn't necessary an optimized node  index (since it's just use the property key to find a saved node index). It should be renamed to avoid confusing.